app/modules/visualization/controllers/visualizationControllers.js

```diff
@@ -2430,6 +2434,12 @@ angular.module('pcApp.visualization.controllers.visualization', [
 					//console.log("-------------->"+arguments[i]['unit_id']);
 					
 					$dataUnit[i] = Unit.getById(arguments[i]['unit_id']);

```

Comment:
Why did you keep the request for unit info even in the case of 0? It produces an error "GET https://services-stage.policycompass.eu/api/v1/references/units/0 404 (NOT FOUND)" at the console. Shouldn't line 2436 (including the promise at line 2444) be inside the else case of the check you added at line 2437?